client/implementation/src/main/java/io/smallrye/graphql/client/typesafe/impl/json/JsonNumberReader.java

```diff
@@ -41,7 +41,7 @@ private Object read(Location location, JsonNumber value, Class<?> rawType) {
             return value.doubleValue();
         if (BigInteger.class.equals(rawType))
             return value.bigIntegerValueExact();
-        if (BigDecimal.class.equals(rawType))
+        if (BigDecimal.class.equals(rawType) || Object.class.equals(rawType))
```

Comment:
Not sure if I understand this correct, but it does not look right that Object maps to BigDecimal ??

Reply:
It's the other way around: the `path` can contain Strings (object field names) and Numbers (array indexes), so it's a `List<Object>`. When we actually receive a number, we could map it to an `Integer`, but as this code could be used by other mappings as well, it's safer to map it to a `BigDecimal` without loosing any precision, etc. As we don't access this number anywhere, only client code looking into the path would be able to tell the difference, and any existing code would have run into an exception before this fix. So it's all safe :-)

Reply:
ok thanks